actors/miner/src/emit.rs

```diff
@@ -0,0 +1,79 @@
+use cid::Cid;
+use fvm_shared::bigint::BigInt;
+use fvm_shared::piece::PaddedPieceSize;
+// A namespace for helpers that build and emit provider Actor events.
+use fil_actors_runtime::runtime::Runtime;
+use fil_actors_runtime::{ActorError, EventBuilder};
+use fvm_shared::sector::SectorNumber;
+
+/// Indicates a sector has been pre-committed.
+pub fn sector_precommitted(rt: &impl Runtime, sector: SectorNumber) -> Result<(), ActorError> {
+    rt.emit_event(
+        &EventBuilder::new().typ("sector-precommitted").field_indexed("sector", &sector).build()?,
+    )
+}
+
+/// Indicates a sector has been activated.
+pub fn sector_activated(
+    rt: &impl Runtime,
+    sector: SectorNumber,
+    unsealed_cid: &Cid,
```

Comment:
Seems reasonable since they're already an `Option` and we force the receiver of the event to special-case these zero-cids too. Are the zero-cids exposed publicly yet at all or are they just an internal detail? Why do we need them like this internally anyway? Are there special cases that require it to be present?

Reply:
They're just precomputed for efficiency. They are public and meaningful as signifying a sector empty of data. There are processes that check for equality with the zero CID